Content.IntegrationTests/Tests/Inventory/ClothingEquipTest.cs

```diff
@@ -0,0 +1,176 @@
+using System.Collections.Generic;
+using System.Linq;
+using Content.IntegrationTests.Tests.Interaction;
+using Content.Server.Actions;
+using Content.Server.Inventory;
+using Content.Server.Spawners.Components;
+using Content.Shared.Actions.Components;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Interaction.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Prototypes;
+using Content.Shared.Storage.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Prototypes;
+
+namespace Content.IntegrationTests.Tests.Inventory;
+
+/// <summary>
+/// This test spawns in every item that has a ClothingComponent and then simulates:
+/// 1. Equipping it (and jumpsuit/hardsuit if needed)
+/// 2. Triggering any InstantActions it grants.
+/// 3. Unequipping it.
+/// </summary>
+public sealed class ClothingEquipTest : InteractionTest
+{
+    private const string HardsuitProto = "ClothingOuterHardsuitEVA";
+    private const string JumpsuitProto = "ClothingUniformJumpsuitColorGrey";
+    protected override string PlayerPrototype => "MobHuman";
+
+    [Test]
+    public async Task EquipAllClothingTesting()
+    {
+        var clothingProtos = ProtoMan.EnumeratePrototypes<EntityPrototype>()
+            .Where(x => x.HasComponent<ClothingComponent>()
+                        && !x.HideSpawnMenu
+                        && !x.Abstract
+                        && !Pair.IsTestPrototype(x)
+                        && !x.HasComponent<RandomSpawnerComponent>()
+                        // Mobs are a massive pain in this test
+                        // (Admeme mouse eating your cak, cancer mouse killing you... mice just dying...)
+                        && !x.HasComponent<MobStateComponent>());
```

Comment:
Previously it was mentioned that it might be good to filter out clothing items that don't do anything, however, it turns out this test also serves as a way to test for the validity of sprites in each possible equip position. This makes me think it might be good to keep this broad-as is, but this is definitely on the slow side of things. It seems to take about 30–40 seconds to run for me, though I think part of that is due to maybe some unneeded RunTicks?